client/src/main/java/io/pravega/keycloak/client/KeycloakAuthzClient.java

```diff
@@ -72,28 +87,24 @@ public String getRPT() {
             // obtain an access token with which to make an authorization request
             AccessTokenResponse accResponse;
             try {
-                accResponse = client.obtainAccessToken();
-                LOG.debug("Obtained access token from Keycloak");
+                accResponse = Retry.withExpBackoff(httpInitialDelayMs, 2, httpMaxRetries)
+                        .retryWhen(isRetryable()).run(() -> client.obtainAccessToken());
             } catch (HttpResponseException e) {
-                LOG.error("Failed to obtain access token from Keycloak", e);
-                if (e.getStatusCode() == HttpStatus.SC_BAD_REQUEST) {
-                    throw new KeycloakAuthenticationException(e);
-                }
-                throw e;
+                LOG.error("Failed to obtain access token from Keycloak");
+                throw new KeycloakAuthenticationException(e);
             }
+            LOG.info("Obtained access token from Keycloak");
```

Comment:
nit: move to inside the try block right after getting the response, so that the log statement is in context. Also, reduce the log level to debug.